src/circuit.rs

```diff
@@ -138,6 +138,6 @@ pub trait Layouter<C: Chip> {
     /// ```
     fn assign_region(
         &mut self,
-        assignment: impl FnOnce(Region<'_, C>) -> Result<(), Error>,
+        assignment: impl FnMut(Region<'_, C>) -> Result<(), Error>,
```

Comment:
This becomes `FnMut` because we call it more than once. Note that we maintain `Region`'s invariant that the inner `to: FnMut`s passed into `Region::assign_advice` and `Region::assign_fixed` are only ever called once.